### PR TITLE
Remove SQLCipher [ci full]

### PR DESCRIPTION
**Pull request by skhamis**
Requirements to land this:
- [x] Remove SQLCipher from build scripts
- [x] Remove SQLCipher migration path
- [x] Tests pass, consumers built/run successfully 
- [x] Update consumers to not use SQLCipher migration path
    - [x] iOS - https://github.com/mozilla-mobile/firefox-ios/pull/16056
    - [x] Fenix - https://github.com/mozilla-mobile/firefox-android/pull/3286

### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [ ] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

File: components/autofill/build.rs [hunk 1]
```diff
@@ -3,6 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
+// TODO: SQLCipher can be removed now, is this comment still relevant?
```
Comment (mhammond) [inline components/autofill/build.rs:6]: I'm fairly sure everything nss related here can die.

File: components/support/rc_crypto/nss/systest/build.rs [hunk 2]
```diff
@@ -6,13 +6,13 @@ use std::{env, ffi::OsString, path::Path};
...
 fn main() {
     // NSS links against SQLite3, so...
-    println!("cargo:rustc-link-lib=static=sqlcipher");
-    println!("cargo:rerun-if-env-changed=SQLCIPHER_LIB_DIR");
-    let sqlcipher_lib_dir: OsString = env::var_os("SQLCIPHER_LIB_DIR").unwrap();
-    println!(
-        "cargo:rustc-link-search=native={}",
-        sqlcipher_lib_dir.to_string_lossy()
-    );
+    // println!("cargo:rustc-link-lib=static=sqlcipher");
```
Comment (mhammond) [inline components/support/rc_crypto/nss/systest/build.rs:9]: all of this can die to I think.

File: components/support/sql/build.rs [hunk 3]
```diff
@@ -3,6 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
+// TODO: Look to see if we can remove this
```
Comment (mhammond) [inline components/support/sql/build.rs:6]: here too

**Comment by skhamis** [comment]
These failures seem unrelated to the patch, so i'll add reviewers for now to get the reviews going and pinky promise those will be green before we land (without significant changes to my code).

**Comment by mhammond** [comment]
This seems fine to me - it looks like a breaking change for consumers, so I assume they have all been updated? Note also that the dependency check failure is related to this PR.

**Comment by skhamis** [comment]
> This seems fine to me - it looks like a breaking change for consumers, so I assume they have all been updated? Note also that the dependency check failure is related to this PR.

thanks for the review! The API calls on the consumers side have been removed (see description for links) so removing it here won't affect any consumers. Updated the dependencies but still figuring out the intermittent issues.

**Comment by skhamis** [comment]
Looks like CI is green!

**Pull request merged**